### ISSUE TITLE
Restoring a deleted file in a deleted subfolder causes unexpected behaviour

### ISSUE THREAD

**Issue by maxemann96** [nextcloud/groupfolders#1257]
**Actual behaviour:**
Restoring a deleted file laying in an also deleted folder restores the file into the root of the groupfolder.

**Expected behaviour:**
The nested directory structure and the file should be restored (with the original permissions of the substructure).

**To reproduce:**
1. Create a group folder `/groupfolder`
2. Create a folder `/groupfolder/sub` with one file `/groupfolder/sub/file`
3. Delete the file `/groupfolder/sub/file`
4. Delete the folder `/groupfolder/sub`
5. Restore the file `/groupfolder/sub/file`

**Comment by pierreozoux** [comment]
I just reproduce, and I agree that it is unexpected.

**Comment by mejo-** [comment]
@maxemann96 and @pierreozoux can you still reproduce this bug with latest version of Groupfolder app on Nextcloud 23? I fail to reproduce the issue on Nextcloud 24 RC1 at least. Also, do you use special storage backends?

**Comment by pierreozoux** [comment]
We are using minio as a storage backend, but no longer using group folder :upside_down_face:

**Comment by mejo-** [comment]
Thanks @pierreozoux, so let's close this issue as non-reproducible for now.

**Comment by maxemann96** [comment]
Hey all, sorry for the very long answer time. This issue still persists. To retest it I created the following structure:
- Create a group folder `/groupfolder`
- Create a sub group folder `/groupfolder/subGroupfolder`
- Create a normal folder `/groupfolder/subGroupfolder/sub` with one file `/groupfolder/subGroupfolder/sub/file`
- Delete the file `/groupfolder/subGroupfolder/sub/file`
- Delete the folder `/groupfolder/subGroupfolder/sub`
- Restore the file `/groupfolder/sub/file`

Notice that the file is restored to `/groupfolder/file`. Which I find is very unexpected that the file gets restored into the root groupfolder. Tested on NC 24.0.1 with group folders version 12.0.0.
